tests/reader/reader_test.go

```diff
@@ -10,70 +12,165 @@ import (
+			Text: tests.SampleText32,
+		}
+
+		contentReader := reader.ContentReader{Config: config}
 
-	result, err := contentReader.ReadFile(file.Name())
-	assert.NoError(t, err)
-	assert.Equal(t, content, result)
+		result, err := contentReader.ReadAll()
+		assert.NoError(t, err)
+		expectedResult := tests.SampleText32 + "\n"
+		assert.Equal(t, expectedResult, result)
+	})
+
+	t.Run("reads multiple files", func(t *testing.T) {
+		fileContents := []string{"Japan", "Australia", "Germany"}
+		expectedResults := strings.Join(fileContents, "\n") + "\n"
+		var filePaths []string
+
+		for index := 0; index < 3; index++ {
+			file, err := tests.CreateTempFile(t, fileContents[index])
+			if err != nil {
+				t.Fatalf("Failed to create temporary file: %s", err)
+			}
+
+			filePaths = append(filePaths, file.Name())
+		}
+
+		config := &options.Config{FilePaths: filePaths}
+
+		contentReader := reader.ContentReader{Config: config}
+
+		result, err := contentReader.ReadAll()
+
+		assert.NoError(t, err)
+		assert.Equal(t, expectedResults, result)
+	})
 }
 
+//nolint:funlen // It cannot be shorter
 func TestReadable(t *testing.T) {
-	file, err := tests.CreateTempFile(t, "content")
-	if err != nil {
-		t.Fatalf("Failed to create temporary file: %s", err)
-	}
-
 	config := &options.Config{}
 	contentReader := reader.ContentReader{Config: config}
 
-	err = contentReader.Readable(file.Name())
-	assert.NoError(t, err)
+	t.Run("file exists and is readable", func(t *testing.T) {
+		file, err := tests.CreateTempFile(t, "content")
+		if err != nil {
+			t.Fatalf("Failed to create temporary file: %s", err)
+		}
 
-	err = contentReader.Readable("nonexistentfile")
-	assert.Error(t, err)
+		err = contentReader.Readable(file.Name())
+		assert.NoError(t, err)
+	})
+
+	t.Run("file does not exist", func(t *testing.T) {
+		err := contentReader.Readable("nonexistentfile")
+		assert.Error(t, err)
+		assert.Equal(t, "file does not exist or can't be accessed", err.Error())
+	})
+
+	t.Run("path is a directory", func(t *testing.T) {
+		dir, err := tests.CreateTempDir(t, "testdir")
+		if err != nil {
+			t.Fatalf("Failed to create temporary directory: %s", err)
+		}
+
+		err = contentReader.Readable(dir)
+		assert.Error(t, err)
+		assert.Equal(t, "reading from directories is not currently supported", err.Error())
+	})
+
+	t.Run("file is not readable", func(t *testing.T) {
+		file, err := tests.CreateTempFile(t, "content")
+		if err != nil {
+			t.Fatalf("Failed to create temporary file: %s", err)
+		}
+
+		// Remove read permissions
+		if err := os.Chmod(file.Name(), 0200); err != nil { // 0200 corresponds to write-only permission.
+			t.Fatalf("Failed to change file permissions: %s", err)
+		}
+
+		err = contentReader.Readable(file.Name())
+		assert.Error(t, err)
+		assert.Equal(t, "permission denied", err.Error())
+
+		// Restore read permissions for cleanup
+		if err := os.Chmod(file.Name(), 0600); err != nil { // 0600 corresponds to read-write permissions.
```

Comment:
```suggestion
		if err := os.Chmod(file.Name(), 0o600); err != nil { // 0o600 corresponds to read-write permissions.
```

You can use the octal notation